### PR TITLE
ref(js): Vastly simplify EventDataSection

### PR DESCRIPTION
**Pull request by evanpurkhiser**
* Props for this component now have comments

 * simplify use of wrapTitle={false}, many usages of this proper were
   disabling it and then "wrapping" the title with a `h3` anyway, which
   is what it did when it was not set to false.

 * Unused `isCentered` prop is removed.

 * EventDataSection has learned to render the QuestionTooltip and wrap a
   GuideAnchor when associated props are passed.

 * Remove the duplicate eventTagsAndScreenshot/dataSection which
   overcomplicated

 * Includes some small visual changes around the alignment of info
   icons, screenshots below

   Before
   

   After

### CODE REVIEW

File: static/app/components/events/eventDataSection.tsx [hunk 1]
```diff
@@ -42,18 +68,27 @@ export function EventDataSection({
   className,
   type,
   title,
-  wrapTitle = true,
+  help,
   actions,
-  isCentered = false,
+  guideTarget,
+  wrapTitle = true,
```
Comment (evanpurkhiser) [inline static/app/components/events/eventDataSection.tsx:74]: I think I can kill this in the future

File: static/app/components/events/eventDataSection.tsx [hunk 2]
```diff
@@ -136,23 +175,18 @@ const SectionHeader = styled('div')<{isCentered?: boolean}>`
...
     & > small {
       margin-left: ${space(1)};
       display: inline-block;
     }
   }
 
-  ${p =>
-    p.isCentered &&
-    css`
-      align-items: center;
```
Comment (evanpurkhiser) [inline static/app/components/events/eventDataSection.tsx:149]: Look above, it's already `align-items: center`

File: static/app/components/events/eventTagsAndScreenshot/dataSection.tsx [hunk 3]
```diff
@@ -1,60 +0,0 @@
...
-
-type Props = {
-  children: React.ReactNode;
-  description: string;
-  title: string;
-  className?: string;
-};
-
-export function DataSection({title, description, children, className, ...props}: Props) {
-  const type = kebabCase(title);
```
Comment (evanpurkhiser) [inline static/app/components/events/eventTagsAndScreenshot/dataSection.tsx:20]: I validated that all the GuideAnchor target's being passed in this way were not used

File: static/app/components/events/eventTagsAndScreenshot/index.tsx [hunk 4]
```diff
@@ -132,13 +132,11 @@ export function EventTagsAndScreenshot({
...
           <ScreenshotWrapper>
             <StyledScreenshotDataSection
+              title={screenshotLink}
+              showPermalink={false}
+              help={t('This image was captured around the time that the event occurred.')}
+              type="screenshot-data-section"
               data-test-id="screenshot-data-section"
-              title={tct('[link]', {
-                link: screenshotLink,
-              })}
```
Comment (evanpurkhiser) [inline static/app/components/events/eventTagsAndScreenshot/index.tsx:138]: 🥵

File: static/app/components/events/interfaces/breadcrumbs/index.tsx [hunk 5]
```diff
@@ -300,10 +300,8 @@ function BreadcrumbsContainer({data, event, organization, projectSlug, isShare}:
   return (
     <EventDataSection
       type={EntryType.BREADCRUMBS}
-      title={<h3>{t('Breadcrumbs')}</h3>}
+      title={t('Breadcrumbs')}
       actions={!showReplay ? actions : null}
-      wrapTitle={false}
-      isCentered
```
Comment (evanpurkhiser) [inline static/app/components/events/interfaces/breadcrumbs/index.tsx:306]: `wrapTitle` already puts it in a `<h3>`, isCentered did nothing.

File: static/app/components/events/interfaces/performance/spanEvidence.tsx [hunk 6]
```diff
@@ -1,9 +1,9 @@
 import styled from '@emotion/styled';
 
+import {EventDataSection} from 'sentry/components/events/eventDataSection';
 import {t} from 'sentry/locale';
 import {EventTransaction, IssueType, Organization} from 'sentry/types';
 
-import {DataSection} from '../../eventTagsAndScreenshot/dataSection';
```
Comment (evanpurkhiser) [inline static/app/components/events/interfaces/performance/spanEvidence.tsx:6]: Having two very similarly named things that kind of do the same thing is dangerous 😬

File: static/app/components/events/eventDataSection.tsx [hunk 7]
```diff
@@ -1,19 +1,45 @@
...
   type: string;
+  /**
+   * Actions that appear to the far right of the title
+   */
   actions?: React.ReactNode;
   className?: string;
-  isCentered?: boolean;
+  /**
+   * If the section has a guide associated to it, you may specify the guide
+   * target and it will wrapp the title with a GuideAnchor
```
Comment (markstory) [inline static/app/components/events/eventDataSection.tsx:27]:
```suggestion
   * target and it will wrap the title with a GuideAnchor
```

**Review by markstory — APPROVED** [review]
Code changes look good to me.

**Pull request merged**